### PR TITLE
Optimise db

### PR DESCRIPTION
**Pull request by vaibhavchellani**

### CODE REVIEW

File: genesis.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "max_tree_depth": 8,
+    "max_tree_depth": 32,
```
Comment (vaibhavchellani) [inline genesis.json:2]: Finally!

File: aggregator/aggregator.go [hunk 2]
```diff
@@ -158,7 +160,18 @@ func (a *Aggregator) processAndSubmitBatch(txs []core.Tx) {
...
+		return commitments, ErrIncorrectTxCount
+	}
+
+	for i := range txs {
+		if i%int(config.GlobalCfg.TxsPerCommitment) == 0 {
+			txsInCommitment = append(txsInCommitment, int(config.GlobalCfg.TxsPerCommitment))
+		}
+	}
+
+	return db.ProcessTxs(&a.LoadedBazooka, &a.DB, txs, txsInCommitment, false)
```
Comment (ChihChengLiang) [inline aggregator/aggregator.go:174]: What are our assumptions here?

Note that 

1. Only the same transaction type can be committed together.
2. Some commitments might be only partially filled.

Comment (vaibhavchellani) [reply aggregator/aggregator.go]: Yeah here we assume we won't partially fill a commitment, the PopTxs() function takes care of only pulling in the right transactions from the pool

File: config/config.go [hunk 3]
```diff
@@ -66,25 +67,25 @@ type Configuration struct {
...
-		Transfer:        ethCmn.Address{}.String(),
-		MassMigration:   ethCmn.Address{}.String(),
-		Create2Transfer: ethCmn.Address{}.String(),
+		DB:                     DefaultDB,
+		DBURL:                  GetDBURL(),
+		Trace:                  false,
+		DBLogMode:              false,
+		EthRPC:                 DefaultEthRPC,
+		TxsPerCommitment:       1,
+		MaxCommitmentsPerBatch: 100,
```
Comment (ChihChengLiang) [inline config/config.go:76]: These two numbers feel not right, why we parameterize them here?
Note that the contract is now parameterized to MaxCommitmentsPerBatch=32

File: db/relayer.go [hunk 4]
```diff
@@ -36,7 +36,15 @@ func (rp *RelayPacket) BeforeCreate(scope *gorm.Scope) error {
 	return nil
 }
 
-func (rp *RelayPacket) AfterCreate(tx *gorm.DB, cfg config.Configuration) (err error) {
+func (rp *RelayPacket) AfterCreate(tx *gorm.DB) (err error) {
+	cfg, err := config.ParseConfig()
```
Comment (ChihChengLiang) [inline db/relayer.go:40]: Great work fixing the cfg arg to the hook.

**Review by ChihChengLiang — APPROVED** [review]
Only take a quick look due to the size of the PR and the urgency we need to cut a release.

File: db/relayer.go (hunk 4 above)
Comment (vaibhavchellani) [reply db/relayer.go:40]: I kinda hate reading the file on every row create but had to do it, for now, will deal better in coming PR's

**Pull request merged**